package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,538 @@
+    quarter_pi, half_pi = np.pi/4, np.pi/2
+    # replace low angles
+    low = screw_angles < quarter_pi
+    screw_angles[low] = np.where(alt_angles[low] < half_pi,
+                                 half_pi - alt_angles[low],
+                                 alt_angles[low] - half_pi)
+    # replace high angles
+    high = screw_angles > 3*quarter_pi
+    screw_angles[high] = np.where(alt_angles[high] < half_pi,
+                                  half_pi + alt_angles[high],
+                                  3*half_pi - alt_angles[high])
+
+    ortho = np.cross(ref_1, rotation_vectors)
+    cos_theta = np.matmul(best_fit, ortho.T)/(pnorm(ortho)*ref_norms[0])
+    bound(cos_theta, -1, 1)
+    angle_to_best_fit = np.arccos(cos_theta)
+    screw_angles[angle_to_best_fit < half_pi] *= -1
+    return np.rad2deg(screw_angles)  # (n_vec,)
+
+
+def helix_analysis(positions, ref_axis=[0, 0, 1]):
+    r"""
+    Calculate helix properties from atomic coordinates.
+
+    Each property is calculated from a sliding window of 4 atoms, 
+    from i to i+3.
+
+    Parameters
+    ----------
+    positions: :class:`numpy.ndarray` of shape (N, 3)
+        Atomic coordinates.
+    ref_axis: array-like of length 3, optional
+        The reference axis used to calculate the tilt of the vector of best fit, 
+        and the local screw angles.
+
+    Returns
+    -------
+    dict with the following keys:
+        local_twists: array, shape (N-3,)
+            local twist angle from atom i+1 to i+2
+        residues_per_turn: array, shape (N-3,)
+            number of residues per turn, based on local_twist
+        local_axes:  array, shape (N-3, 3)
+            the length-wise helix axis of the local window
+        local_bends: array, shape (N-6,)
+            the angles between local helix angles, 3 windows apart
+        local_heights: array, shape (N-3,)
+            the rise of each local helix
+        local_rotation_vectors: array, shape (N-2, 3)
+            the unit vector from each local origin to atom i+1
+        local_origins: array, shape (N-2, 3)
+            the projected origin for each helix
+        all_bends: array, shape (N-3, N-3)
+            angles between each local axis
+        global_axes: array, shape (3,)
+            vector of best fit through origins
+        local_screw: array, shape (N-2,)
+            angles of each rotation vector to normal plane of global axes
+    """
+
+    #          ^               ^
+    #           \             / bi
+    #            \           /
+    #         CA_i+2 <----- CA_i+1
+    #         /    \       /   ^
+    #        /    r \     /     \
+    #     V /        \ θ /       \
+    #      /          \ /       CA_i
+    #     v           origin
+    #   CA_i+3
+    #
+    # V: vectors
+    # bi: bisectors
+    # θ: local_twists
+    # origin: origins
+    # local_axes: in the plane of the screen. Orthogonal to the bisectors
+
+    vectors = positions[1:] - positions[:-1]  # (n_res-1, 3)
```

Comment:
maybe a silly question, what's the anticipated size of `positions` here?

Reply:
I think helices are usually < 50 residues long so ``(50, 3)`` for ideal use, but there are no error-raising checks on atom groups so you could just put your entire trajectory into it.